--- old/dynamic_dns_updater.sh	2017-08-27 21:55:32.347947900 +0800
+++ new/dynamic_dns_updater.sh	2017-08-27 21:53:25.019258454 +0800
@@ -83,6 +83,8 @@ if [ "$ip_source" = "script" ]; then
 	[ -z "$1" ] && write_log 14 "No script defined to detect local IP!"
 	[ -x "$1" ] || write_log 14 "Script to detect local IP not executable!"
 fi
+#set domain_script
+set -- $domain_script
 get_seconds CHECK_SECONDS ${check_interval:-10} ${check_unit:-"minutes"}
 get_seconds FORCE_SECONDS ${force_interval:-72} ${force_unit:-"hours"}
 get_seconds RETRY_SECONDS ${retry_interval:-60} ${retry_unit:-"seconds"}
