--- old/wanip	1970-01-01 08:00:00.000000000 +0800
+++ new/wanip	2017-09-16 12:32:38.924617466 +0800
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+curl -s http://1212.ip138.com/ic.asp | sed -ne "s/.*\[\(.*\)\].*$/\1/p" 2>/dev/null
+#upnpc -s | grep "ExternalIPAddress" | sed 's/[^0-9\.]//g'
+
+#myip:
+#http://1212.ip138.com/ic.asp
+#.*IP.*[ip].*$
+#curl -s http://checkip.dns.he.net
+#^Your IP address is : 101.248.255.208</body>$
+#https://submit.apnic.net/templates/yourip.html
+#^[<strong>ip</strong>]$
+#https://ipinfo.info/ipn/
+#<h4>203.104.193.100<br>
